src/Kunstmaan/MediaBundle/Helper/MediaManager.php

```diff
@@ -160,4 +184,26 @@ public function getFolderAddActions()
 
         return $result;
     }
+
+    public function isExtensionAllowed(File $file): bool
+    {
+        if (!$this->limitAllowedExtesions) {
+            return true;
+        }
+
+        $mimeType = $this->mimeTypes->guessMimeType($file->getRealPath());
+        $extensions = $this->mimeTypes->getExtensions($mimeType);
+        $extension = array_shift($extensions);
+        if (!in_array($extension, $this->allowedExtensions, true)) {
```

Comment:
The code itself looks ok, but it yields unexpected results. Forexample if I only allow .txt files to be uploaded and I upload a .json file, this checks allows it.
This is because `$this->mimeTypes->guessMimeType()` for a json files returns `text/plain`, which results in `"txt", "text","conf","def","list","log","in","asc"` as extensions from `$this->mimeTypes->getExtensions` and so is `.txt` picked. And the upload is allowed. Although there is technically no bug in this code, the result is unexpected.

So we should think about a solution so that allowed extensions list is actually the only extensions allowed

Reply:
Ik heb een soort van fallback gemaakt waarbij we kijken dat indien de locale fileinfo text/plain teruggeeft en het een uploadedfile is we fallbacken naar de clientmimetype. Dat is natuurlijk niet 100% foolproof want is spoofable maar at least kunnen we toch iets meer onderscheid maken tussen al die files die anders als "text/plain" gewoon zomaar toegelaten worden.

Reply:
I'm not sure about this update as symfony indicates (and you) this value can't be trusted. https://github.com/symfony/symfony/blob/0baa58d4e4bb006c4ae68f75833b586bd3cb6e6f/src/Symfony/Component/HttpFoundation/File/UploadedFile.php#L128-L129

Would it be a solution to move away from the idea of listing extensions and have the user provide an allowed list of mimetypes? Like you would do with the [File constraint](https://symfony.com/doc/current/reference/constraints/File.html#mimetypes)?

Reply:
The thing is that its just a server thing. because of the way the upload works the uploaded file will be put in the temp folder as "9d0329d30" or something like that without an extension. Because files like json,xml etc are not executable nor have a header in the file to specify what file type they are this effectively makes them just a plaintext file. The good thing is that because of that they are not a threat to the server as they wouldn't be able to be executed. Therefore i think its more a matter of cleanliness to do this optional extra check for files we have already verified are not a threat to the system to deny them from being uploaded if we have not allowed that extension.

Using mimetypes would give us the exact same "problem" because someone might be confused as to why a file with mimetype "text/json" is allowed on the server if he hasn't put it in his allowed configuration.

But yeah tldr: files that get detected by fileinfo on the server as plaintext are no threat and therefore its just a matter of cleanliness to choose to allow/block them from being uploaded which we can do using the clientinfo.